### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by ANazaret**

### CODE REVIEW

File: src/treeffuser/scaler.py [hunk 1]
```diff
@@ -3,65 +3,70 @@
...
+        Parameters
+        ----------
+        X : ndarray of shape (batch, x_dim)
+            The data to fit the preprocessor to.
+        cat_idx : list of int, optional
+            The indices of the categorical features.
         """
         self._reset()
         cat_idx = cat_idx if cat_idx is not None else []
+        non_cat_idx = X[:, [i for i in range(X.shape[1]) if i not in cat_idx]]
```
Comment (velezbeltran) [inline src/treeffuser/scaler.py:49]: Naming here is a bit confusing. It gives the impression that `cat_idx` and `non_cat_idx` should be complements of one another but they are really not.

File: testbed/src/testbed/models/lightning_uq_models/_card.py [hunk 2]
```diff
@@ -26,10 +26,10 @@
 from torch.optim import Adam
 from tqdm import tqdm
 
-from testbed.models._preprocessors import Preprocessor
 from testbed.models.base_model import ProbabilisticModel
 from testbed.models.lightning_uq_models._data_module import GenericDataModule
 from testbed.models.lightning_uq_models._utils import _to_tensor
+from treeffuser.scaler import ScalerMixedTypes
```
Comment (velezbeltran) [inline testbed/src/testbed/models/lightning_uq_models/_card.py:32]: I am still a bit unsure about how to handle this situation where we depend on Treeffuser in a way that is beyond the Treeffuser class itself. I am not sure I like this but I don't like the solution that was there before of duplicating code. 
Just pointing it out.

**Review by velezbeltran — CHANGES_REQUESTED** [review]
Thanks looks good! Only left one thing to change and a general comment.

File: src/treeffuser/scaler.py (hunk 1 above)
Comment (ANazaret) [reply src/treeffuser/scaler.py:49]: Good catch, thanks!

**Review by velezbeltran — APPROVED** [review]

**Pull request merged**